pkg/config/loader/config_entry_loader.go

```diff
@@ -55,7 +56,19 @@ func parseConfigEntry(
 		return nil, []error{newDefinitionParserError(configId, singleConfigContext, err.Error())}
 	}
 
+	for _, group := range definition.GroupOverrides {
```

Comment:
I would move each check into a separate function to decrease the complexity and have an opportunity to comment on what is checked under which assumption.

My opinion you decide if it makes sense.

Reply:
I have also improved the actual log message.